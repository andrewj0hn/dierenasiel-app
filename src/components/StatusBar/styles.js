// @flow
import { StyleSheet } from 'react-native';
import { STATUSBAR_COLOR } from '../../constants/Styles';

export default StyleSheet.create({
  statusBarHeight: {
    backgroundColor: STATUSBAR_COLOR,
  },
});
