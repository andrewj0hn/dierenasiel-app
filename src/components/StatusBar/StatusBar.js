// @flow
import React from 'react';
import { StatusBar as StatusBarRN, View } from 'react-native';
import styles from './styles';
import { STATUS_BAR_STYLE } from '../../constants/Styles';

type Props = {
  height: number,
};

const defaultProp = {
  height: 0,
};

const StatusBar = ({ height }: Props) => (
  <View>
    <StatusBarRN barStyle={STATUS_BAR_STYLE} />
    <View style={[styles.statusBarHeight, { height }]} />
  </View>
);

StatusBar.defaultProp = defaultProp;

export default StatusBar;
