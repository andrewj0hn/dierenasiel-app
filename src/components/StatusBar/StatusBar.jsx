// @flow
import React from 'react';
import { StatusBar as StatusBarRN, View } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';
import { STATUS_BAR_STYLE } from '../../constants/Styles';

const StatusBar = ({ height }) => (
  <View>
    <StatusBarRN barStyle={STATUS_BAR_STYLE} />
    <View style={[styles.statusBarHeight, { height }]} />
  </View>
);

StatusBar.propTypes = {
  height: PropTypes.number,
};

StatusBar.defaultProps = {
  height: 0,
};

export default StatusBar;
