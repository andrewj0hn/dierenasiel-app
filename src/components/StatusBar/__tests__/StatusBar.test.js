import React from "react";
import StatusBar from "../StatusBar";
import renderer from "react-test-renderer";

test("renders correctly", () => {
    const tree = renderer.create(<StatusBar/>).toJSON();
    expect(tree).toMatchSnapshot();
});