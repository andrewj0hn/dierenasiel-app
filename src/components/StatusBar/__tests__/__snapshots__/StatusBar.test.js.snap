// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "green",
        },
        Object {
          "height": 0,
        },
      ]
    }
  />
</View>
`;
